### PR TITLE
TEAM TEACHING - Primitive entities render in batch

### PR DESCRIPTION
**Pull request by Atlante45**
This PR moves a lot of the entities rendering to the new render pipeline.

* DeferredLighting simple program takes a batch
* Removed used DeferredLighting methods not using a batch
* Removed glDisableClientState calls from GeometryCache
* GLBackend push/pops client attribs and sets _attributeActivation accordingly
* Replaced pos/rot/dim by transform in EntityItem
* Added batch in RenderArgs
* Entity proxies render in entities batch
* Box, EntityDebug, Light, Line, Sphere, Zones:
      - Removed glPush/Pop calls
      - Render in entities batch
* 3D Overlays render before entities

I am now working on getting Particles, Text and Web entities to render in the entities batch.
It's a bit more tricky as the all use either a textured geometry or a third party renderer that has to be moved to using a batch.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: a9a0c9f387b52e38e0ba441da4cd8d991468021f available at: https://gist.github.com/e1d8766c720433bcf24c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4904.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4904.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e67622d21997cbb3809e3a6dad254e8ade67a907 available at: https://gist.github.com/e1d8766c720433bcf24c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4904.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4904.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: eef87412fd4ea8fa9253b2ca08b0fa8a3486ac26 available at: https://gist.github.com/e1d8766c720433bcf24c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4904.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4904.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9b49936d2930106b3ab4a67ed13a0385f8f07965 available at: https://gist.github.com/e1d8766c720433bcf24c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4904.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4904.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9a007ab1356c83f317dfae0fe943ae5778d93c17 available at: https://gist.github.com/e1d8766c720433bcf24c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4904.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4904.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9c9aee667af1b41adc12fcb7f619e1dea3c72b86 available at: https://gist.github.com/e1d8766c720433bcf24c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4904.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4904.dmg

File: libraries/entities-renderer/src/RenderableDebugableEntityItem.cpp [hunk 1]
```diff
@@ -21,48 +24,24 @@
...
-            DependencyManager::get<DeferredLightingEffect>()->renderSolidSphere(0.5f, SLICES, STACKS, blueColor);
-        glPopMatrix();
-    glPopMatrix();
+    glm::vec4 blueColor(0.0f, 0.0f, 1.0f, 1.0f);
+    
+    Q_ASSERT(args->_batch);
+    gpu::Batch& batch = *args->_batch;
+    Transform transform = entity->getTransformToCenter();
+    // Cancel true dimensions and set scale to 2 * radius (diameter)
+    transform.postScale(2.0f * glm::vec3(radius, radius, radius) / entity->getDimensions());
```
Comment (jherico) [inline libraries/entities-renderer/src/RenderableDebugableEntityItem.cpp]: glm::vec3(radius) is equivalent to glm::vec3(radius, radius, radius);

**Pull request merged**